src/shaders/metallic-roughness.frag

```diff
@@ -341,43 +382,85 @@ void main()
     // Anything less than 2% is physically impossible and is instead considered to be shadowing. Compare to "Real-Time-Rendering" 4th editon on page 325.
     vec3 specularEnvironmentR90 = vec3(clamp(reflectance * 50.0, 0.0, 1.0));
 
+    // LIGHTING
+
+    vec3 color = vec3(0.0, 0.0, 0.0);
+    vec3 normal = getNormal();
+    vec3 view = normalize(u_Camera - v_Position);
+
     MaterialInfo materialInfo = MaterialInfo(
         perceptualRoughness,
         specularEnvironmentR0,
         alphaRoughness,
         diffuseColor,
         specularEnvironmentR90,
-        specularColor
+        specularColor,
+        normal,
+        0.0 //Clearcoat factor is null
+    );
+    MaterialInfo clearCoatInfo = MaterialInfo(
+        clearcoatRoughness,
+        vec3(0.04), //fixed from specification //todo use variable from functions
+        clearcoatRoughness * clearcoatRoughness, //alphaRoughness is roughnessvalue squared
+        vec3(0.0), //clearcoat layer has no diffuse color
+        vec3(1.0), //F_90 = 1.0
+        vec3(1.0), //clearcoat layer color is white
+        clearcoatNormal,
+        clearcoatFactor
     );
-
-    // LIGHTING
-
-    vec3 color = vec3(0.0, 0.0, 0.0);
-    vec3 normal = getNormal();
-    vec3 view = normalize(u_Camera - v_Position);
 
 #ifdef USE_PUNCTUAL
     for (int i = 0; i < LIGHT_COUNT; ++i)
     {
+        vec3 lightColor = vec3(0);
+        vec3 clearcoatColor = vec3(0);
+        vec3 pointToLight = vec3(0);
         Light light = u_Lights[i];
         if (light.type == LightType_Directional)
         {
-            color += applyDirectionalLight(light, materialInfo, normal, view);
+            lightColor = applyDirectionalLight(light, materialInfo, view);
+            #ifdef MATERIAL_CLEARCOAT
+                pointToLight = -light.direction;
+                clearcoatColor = applyDirectionalLight(light, clearCoatInfo, view);
+            #endif
         }
         else if (light.type == LightType_Point)
         {
-            color += applyPointLight(light, materialInfo, normal, view);
+            lightColor = applyPointLight(light, materialInfo, view);
+            #ifdef MATERIAL_CLEARCOAT
+                pointToLight = light.position - v_Position;
+                clearcoatColor = applyPointLight(light, clearCoatInfo, view);
+            #endif
         }
         else if (light.type == LightType_Spot)
         {
-            color += applySpotLight(light, materialInfo, normal, view);
+            lightColor = applySpotLight(light, materialInfo, view);
+            #ifdef MATERIAL_CLEARCOAT
+                pointToLight = light.position - v_Position;
+                clearcoatColor = applySpotLight(light, clearCoatInfo, view);
+            #endif
         }
+        #ifdef MATERIAL_CLEARCOAT
+            AngularInfo angularInfo = getAngularInfo(pointToLight, clearCoatInfo.normal, view);
```

Comment:
`getAngularInfo` is also called inside the `apply...Light` functions so we should extract it and only call it once